webapp/lib/controller.dart

```diff
@@ -275,38 +329,38 @@ void fetchConfigurationData() {
   configurationResponseLanguages = configurationReponseLanguageData;
 }
 
-Map<String, bool> getTagList(String selectedTag, Map<String, Map<String, List<String>>> tagData) {
+Map<String, bool> getTagList(String selectedTag, Map<String, List<List<String>>> tagData) {
   return new Map.fromIterable(tagData.keys.toList(),
     key: (tag) => tag,
     value: (tag) => selectedTag != null && selectedTag == tag ? true : false);
 }
 
-void populateConfigurationView(String selectedTag, Map<String, bool> tagList, Map<String, List<String>> tagResponses) {
+void populateConfigurationView(String selectedTag, Map<String, bool> tagList, List<String> responseLanguages, List<List<String>> tagResponses) {
   view.contentView.configurationView.tagList.renderTagList(tagList);
-  view.contentView.configurationView.tagResponses.renderResponses(selectedTag, tagResponses);
+  view.contentView.configurationView.tagResponses.renderResponses(selectedTag, responseLanguages, tagResponses);
 }
 
-void addNewConfigurationTag(String tagToAdd, List<String> availableLanguages, Set<String> additionalTags, Map<String, Map<String, List<String>>> tagData) {
-  tagData[tagToAdd] = new Map.fromIterable(availableLanguages, key: (d) => d, value: (d) => ['']);
+void addNewConfigurationTag(String tagToAdd, List<String> availableLanguages, Set<String> additionalTags, Map<String, List<List<String>>> tagData) {
+  tagData[tagToAdd] = [['','']];
```

Comment:
should this list have as many elements as there are in the `availableLanguages` list?

Reply:
I agree. Refactored it.